cmd/terramate/cli/cli.go

```diff
@@ -645,38 +645,35 @@ func (c *cli) checkGitUncommited() bool {
 	return true
 }
 
-func (c *cli) gitSafeguards(checks terramate.RepoChecks, shouldAbort bool) {
-	logger := log.With().
-		Str("action", "gitSafeguards()").
-		Logger()
+func debugFiles(files []string, msg string) {
+	for _, file := range files {
+		log.Debug().
+			Str("file", file).
+			Msg(msg)
+	}
+}
 
+func (c *cli) gitSafeguards(checks terramate.RepoChecks, shouldAbort bool) {
 	if c.parsedArgs.Run.DryRun {
 		return
 	}
 
-	if c.checkGitUntracked() && len(checks.UntrackedFiles) > 0 {
-		if shouldAbort {
-			logger.Fatal().
-				Strs("files", checks.UntrackedFiles).
-				Msg("repository has untracked files")
-		} else {
-			logger.Warn().
-				Strs("files", checks.UntrackedFiles).
-				Msg("repository has untracked files")
-		}
+	debugFiles(checks.UntrackedFiles, "untracked file")
+	debugFiles(checks.UncommittedFiles, "uncommitted file")
+
+	var logger *zerolog.Event
+	if shouldAbort {
+		logger = log.Fatal()
+	} else {
+		logger = log.Warn()
+	}
 
+	if c.checkGitUntracked() && len(checks.UntrackedFiles) > 0 {
+		logger.Msg("repository has untracked files")
```

Comment:
hmm I don't like this approach much because now this unsuspicious logging can abort... the other way it was explicit when it aborts... but maybe it's just me.

Reply:
hmm the logger is created pretty close to here and you can see clearly when it is fatal and warning, maybe if the function was bigger and had more stuff on it but on this case it seemed OK to me. but can change it too, no hard feelings on this.

Reply:
better to be explicit, take a look now :-)